### PR TITLE
chore: Add semantic-release for automatic package releases

### PR DESCRIPTION
**Pull request by djones**
This adds https://github.com/semantic-release/semantic-release support.

The default plugins that run include GitHub and NPM and I've configured CircleCI to have the correct tokens.

It's a wee bit hard to actually test this all works but the release step does seem to [run and bail out](https://circleci.com/gh/percy/percy-agent/1118?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) since it's not on master, which is to be expected.

From this point forward our pull request titles will need to [have certain keywords in them](https://semantic-release.gitbook.io/semantic-release/#commit-message-format).

### CODE REVIEW

**Comment by Robdel12** [comment]
I think this might be a `chore`? I wonder if we want to setup a `next` branch we merge into just so chore or not code changes aren't released to NPM 🤔

**Comment by djones** [comment]
Agreed, changed to "chore" prefix @Robdel12.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+```
+4. Make your code changes, following the [Coding rules](#coding-rules)
+5. Push your topic branch up to your fork:
+```bash
+$ git push origin <topic-branch-name>
+```
+6. [Open a Pull Request](https://help.github.com/articles/creating-a-pull-request/#creating-the-pull-request) with a clear title and description.
+
+**Tips**:
+- For ambitious tasks, open a Pull Request as soon as possible with the `[WIP]` prefix in the title, in order to get feedback and help from the community.
```
Comment (Robdel12) [inline CONTRIBUTING.md]: Can forks create drafts? Might be worth updating that here 😃

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+$ npm run test
+```
+
+You can test Percy Agent Client with
+
+```bash
+$ npm run test-client
+```
+
+And the integration tests can be run with:
```
Comment (Robdel12) [inline CONTRIBUTING.md]: Might be worth mentioning they'll need to set a `PERCY_TOKEN`?

File: RELEASING.md [hunk 3]
```diff
@@ -1,17 +1,3 @@
...
-1. `git push --tags`
-1. Ensure tests have passed on that tag
-1. [Update the release notes](https://github.com/percy/percy-agent/releases) on GitHub
-1. `npm publish --access=public` (leave new version blank)
-1. [Visit npm](https://www.npmjs.com/package/@percy/agent) and see the new version is live
-
-* Announce the new release,
-   making sure to say "thank you" to the contributors
-   who helped shape this version!
+Releasing of the `@percy/agent` package is handled automatically with the [semantic-release](https://github.com/semantic-release/semantic-release) package.
```
Comment (Robdel12) [inline RELEASING.md]: 🔥

**Pull request merged**